### ISSUE TITLE
Writing Jest Tests for Typescript files that Import strip-html

### ISSUE THREAD

**Issue by jrock2004** [codsen/codsen#43]
**Package's name**
string-strip-html: version `9.1.6`

**Describe the situation**
I have a node typescript project that I am trying to write jest test for. When I run the test it bombs out because it says I can import statement outside a module

![image](https://user-images.githubusercontent.com/655716/159104804-893a6872-7051-4288-a4a4-b0e3e4f577e1.png)

// tsconfig

```json
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true
  }
}
```

// jest.config.ts

```ts
export default {
  moduleFileExtensions: ['js', 'ts', 'tsx'],
  transform: {
    '^.+\\.(ts|tsx)$': 'ts-jest',
  },
  testMatch: ['**/tests/**/*.spec.ts', '**/tests/**/*.test.ts'],
  testEnvironment: 'node',
};
```

Thoughts?

**Comment by revelt** [comment]
hi mate! Sorry about the trouble. So, `moment` is not pure ESM package yet, you can see they use `main` in [package.json](https://github.com/moment/moment/blob/develop/package.json) instead of `exports`+`type: module`.

Try https://kulshekhar.github.io/ts-jest/docs/guides/esm-support/ first. 

If all fails or you need a solution quickly, roll back to v8 of `string-strip-html` — the `8.3.0` which is the last version before we migrated to ESM. When the tooling matures, you can always bump to v9.

**Comment by jrock2004** [comment]
Yeah so it seems like there are other plugins that have same issues. When I downgraded to version `8.3.0` it fix this plugins issue, but then other plugin had same issue. Guess I am going to wait on this or maybe move on from imports. Not sure, but thanks

**Comment by avlapeto** [comment]
Hello!
Same issue with `13.0.6`

**Comment by revelt** [comment]
hi all! I back-ported 13.0.3 to CJS and released as 8.4.0 — the 8.4.0 is a bundled CJS build, no ESM. It does not consume any dependencies in package.json — they are all transpiled to CJS and bundled in. Try it.

**Comment by Goolyio** [comment]
> Hello! Same issue with `13.0.6`
> 
> Resolved by processing `*.esm.js` modules via `ts-jest`:
> 
> ```
>   extensionsToTreatAsEsm: ['.ts', '.esm.js'],
>   transform: {
>     '^.+\\.(ts|js)$': 'ts-jest',
>   },
>   transformIgnorePatterns: ['/node_modules/(?!dot-prop).+(?<!\\.esm)\\.js$'],
> ```
> 
> `tsconfig.jsons`:
> 
> ```
> ...
> "ts-node": {
>     "esm": true,
>     "experimentalSpecifierResolution": "node"
>   }
> ```

Been trying to resolve this for days, this is the only thing that worked. Thanks @avlapeto ❤️.
